### PR TITLE
feat(m2): enforce linux seccomp loading with fail-closed behavior

### LINKED ISSUE #121: Linux enforcement: implement seccomp filter loading with fail-closed behavior
## Context
Parent tracker: #69

Seccomp capability probing exists, but runtime seccomp enforcement is still effectively no-op.

## Scope
Implement seccomp filter materialization/loading in Linux launch path with explicit fail-closed semantics.

## Notes
- Keep the policy minimal and aligned to current alpha command model.
- Document compatibility assumptions and fallback boundaries clearly.

## Done When
- Linux runtime loads and enforces seccomp filter(s) before target exec.
- Launch fails explicitly if seccomp cannot be applied where required.
- Tests verify enforcement path and failure semantics.
- `cargo fmt`, `cargo clippy`, and `cargo test` pass for impacted crates.

Related: #3, #69

### PR DESCRIPTION
**Pull request by manuelpenazuniga**
## Summary
- add Linux seccomp filter materialization using seccompiler
- install seccomp in child pre_exec after rlimits and Landlock
- fail closed on seccomp build/compile/apply errors (spawn returns error)
- deny high-risk syscalls by default and block socket syscalls when net=none
- add Linux-only tests for seccomp deny (net=none) and allow (net=open) behavior

## Validation
- cargo fmt --all
- cargo clippy -p clawcrate-sandbox --all-targets -- -D warnings
- cargo test -p clawcrate-sandbox
- cargo check -p clawcrate-sandbox --target x86_64-unknown-linux-gnu
- cargo clippy --workspace --all-targets -- -D warnings
- cargo test --workspace

Closes #121

### CODE REVIEW

File: crates/clawcrate-sandbox/src/linux.rs [hunk 1]
```diff
@@ -583,10 +664,37 @@ fn apply_linux_landlock_restrictions(context: &LinuxLandlockContext) -> io::Resu
...
+#[allow(unsafe_code)]
+fn apply_linux_seccomp_filter(context: &LinuxSeccompContext) -> io::Result<()> {
+    // SAFETY: prctl contract is satisfied for PR_SET_NO_NEW_PRIVS.
+    let prctl_result = unsafe { libc::prctl(libc::PR_SET_NO_NEW_PRIVS, 1, 0, 0, 0) };
+    if prctl_result != 0 {
+        return Err(io::Error::from_raw_os_error(Errno::last_raw()));
+    }
+
+    seccompiler::apply_filter(context.program.as_slice())
+        .map_err(|source| io::Error::other(format!("failed to apply seccomp filter: {source}")))
```
[the remark this answers is not in the document]
Comment (manuelpenazuniga) [reply crates/clawcrate-sandbox/src/linux.rs:689]: @manuelpenazuniga check that!

File: crates/clawcrate-sandbox/src/linux.rs [hunk 2]
```diff
@@ -389,9 +465,14 @@ impl LinuxSandbox {
         let landlock_context =
             prepare_linux_landlock_context(prepared).map_err(LinuxSandboxError::Spawn)?;
```
[the remark this answers is not in the document]
Comment (manuelpenazuniga) [reply crates/clawcrate-sandbox/src/linux.rs:466]: ok
